In [1]:
import pandas as pd
import scipy.stats as stats
import matplotlib.pyplot as plt
%matplotlib inline

## Franco's Putsch und Lebensspanne der AutorInnen

Haben die AutorInnen, die kurz nach dem Putsch und dem Bürgerkrieg gestorben sind, ein statistisch kürzeres Leben als die andere AutorInnen gehabt?

In [5]:
sp_authors = pd.read_csv("../Datensaetze/sp-authors.tsv", sep="\t")

In [6]:
sp_authors.head()

id                     author-fullname      author  birth  death  gender  \
0   1                   Rosalía de Castro    RdCastro   1837   1885  female   
1   2               Enrique Pérez Escrich     Escrich   1829   1897    male   
2   3  Antonio de Trueba y de la Quintana      Trueba   1819   1889    male   
3   4                José Selgas Carrasco      Selgas   1822   1882    male   
4   5        Francisco Navarro Villoslada  Villoslada   1818   1895    male   

   pages-in-manual  novels  works  life-span  digitzed-by  
0               17       1      1         48            6  
1                1       4     17         68            5  
2                1       0      1         70            6  
3                1       5      7         60            4  
4                1       0      0         77            5

In [7]:
sp_authors["life-span"].mean()

67.69924812030075

In [8]:
sp_authors.shape

(133, 11)

In [9]:
dead_authors_war = sp_authors.loc[ (sp_authors["death"] >= 1936) & (sp_authors["death"]<= 1945)]

In [10]:
dead_authors_war.shape

(26, 11)

In [11]:
dead_authors_non_war = sp_authors.loc[ (sp_authors["death"] < 1936) | (sp_authors["death"]> 1945)]

In [12]:
dead_authors_non_war.shape

(107, 11)

In [13]:
dead_authors_war["life-span"].mean()

60.0

In [14]:
dead_authors_non_war["life-span"].mean().round(1)

69.6

Die AutorInnen, die während und kurz nach dem Krieg gestorben sind, lebten 60 in Durchsnitt. Die andere AutorInnen 70 Jahre. Ist der Unterschied von 10 Jahren statistisch signifikant? D. h. wie wahrscheinlich ist dieser  Unterschied unter Zufall?

Um diese Antwort zu beantworten, verwenden wir den T-test:


### Independent two sample t test

https://en.wikipedia.org/wiki/Student%27s_t-test#Independent_two-sample_t-test

<img src="https://wikimedia.org/api/rest_v1/media/math/render/svg/070d365e1b347ea5f83f0147043868fa120b6646" width="100"/>
where:

<img src="https://wikimedia.org/api/rest_v1/media/math/render/svg/f63e99c30c01445668ea2cce9832da0f6810cb4b" width="120"/>

In [15]:
x1 = dead_authors_war["life-span"].mean()
x2 = dead_authors_non_war["life-span"].mean()


In [26]:
s2x1 = dead_authors_war["life-span"].var()
s2x2 = dead_authors_non_war["life-span"].var()
print(s2x1,s2x2)

139.36 222.9266443308058


In [28]:
sp = (( (s2x1) + s2x2) / 2) ** 0.5
sp

13.458949519386827

In [29]:
n = sp_authors.shape[0]
n

133

In [30]:
t = (x1 - x2) / (sp * ( ( 2 / n )**0.5 ) )

In [31]:
t

-5.798502348421776

In [33]:
df = n- 1
df

132

Durch eine T-Tabelle kann man aus dem Betrag des T-Werts, gegeben Degrees of freedom (df), den P-Wert für two-tails Test. Für einen T-Wert von 5.79, 132 degrees of freedom, der P-Wert ist kleiner als 0.001.

<img src="https://cdn-images-1.medium.com/max/1600/1*0_EW4xt4le_SIrmO7z9FPA.png" width="700"/>

Das heißt, die AutorInnen, die während und kurz nach dem Krieg gestorben sind, sind jünger gestorben, u.z. 10 Jahre jünger, und dieser Unterschied ist statistisch signifikant (P-Wert < 0.001).